src/transformers/pipelines/token_classification.py

```diff
@@ -210,63 +244,124 @@ def __call__(self, inputs: Union[str, List[str]], **kwargs):
         _inputs, offset_mapping = self._args_parser(inputs, **kwargs)
         if offset_mapping:
             kwargs["offset_mapping"] = offset_mapping
-
         return super().__call__(inputs, **kwargs)
 
-    def preprocess(self, sentence, offset_mapping=None):
+    def preprocess(self, sentence, offset_mapping=None, **preprocess_params):
+        tokenizer_params = preprocess_params.pop("tokenizer_params", {})
         truncation = True if self.tokenizer.model_max_length and self.tokenizer.model_max_length > 0 else False
-        model_inputs = self.tokenizer(
+        inputs = self.tokenizer(
             sentence,
             return_tensors=self.framework,
             truncation=truncation,
             return_special_tokens_mask=True,
             return_offsets_mapping=self.tokenizer.is_fast,
+            **tokenizer_params,
         )
-        if offset_mapping:
-            model_inputs["offset_mapping"] = offset_mapping
+        num_chunks = len(inputs["input_ids"])
 
-        model_inputs["sentence"] = sentence
+        for i in range(num_chunks):
+            if self.framework == "tf":
+                model_inputs = {k: tf.expand_dims(v[i], 0) for k, v in inputs.items()}
+            else:
+                model_inputs = {k: v[i].unsqueeze(0) for k, v in inputs.items()}
+            if offset_mapping:
+                model_inputs["offset_mapping"] = offset_mapping if i == 0 else None
+            model_inputs["sentence"] = sentence if i == 0 else None
+            model_inputs["is_last"] = i == num_chunks - 1
 
-        return model_inputs
+            yield model_inputs
 
     def _forward(self, model_inputs):
         # Forward
-        special_tokens_mask = model_inputs.pop("special_tokens_mask")
-        offset_mapping = model_inputs.pop("offset_mapping", None)
-        sentence = model_inputs.pop("sentence")
+        model_outputs = model_inputs
+        model_inputs = {k: model_inputs[k] for k in self.tokenizer.model_input_names}
         if self.framework == "tf":
-            logits = self.model(model_inputs.data)[0]
+            logits = self.model(**model_inputs)[0]
         else:
             output = self.model(**model_inputs)
             logits = output["logits"] if isinstance(output, dict) else output[0]
+        model_outputs["logits"] = logits
 
-        return {
-            "logits": logits,
-            "special_tokens_mask": special_tokens_mask,
-            "offset_mapping": offset_mapping,
-            "sentence": sentence,
-            **model_inputs,
-        }
+        return model_outputs
 
-    def postprocess(self, model_outputs, aggregation_strategy=AggregationStrategy.NONE, ignore_labels=None):
+    def postprocess(
+        self,
+        all_outputs,
+        aggregation_strategy=AggregationStrategy.NONE,
+        ignore_labels=None,
+        **postprocess_params,
```

Comment:
If tokenizer doesn't have `max_length` then the `stride` argument shouldn't be used because the model doesn't require it, it can do without chunking.

`stride=None` is perfectly correct for this default here.

For the unlimited lengthe tokenizer, we should raise an error instead. (Maybe chunking could still be desired for performance reasons in which case I think users should artificially limit the tokenizer for this to work, wdyt ?)